studio/src/aem/aem-tag-picker-field.js

```diff
@@ -0,0 +1,293 @@
+
+const AEM_TAG_PATTERN = /^[a-zA-Z][a-zA-Z0-9]*:/;
+const namespaces = {};
+
+/**
+ * Converts from attribute (tag format) to property (path format)
+ * @param {string} value Tag format string (e.g., "mas:product/photoshop")
+ * @returns {string} Path format string (e.g., "/content/cq:tags/mas/product/photoshop")
+ */
+export function fromAttribute(value) {
+    if (!value) return [];
+    const tags = value.split(',');
+    return tags
+        .map((tag) => tag.trim())
+        .map((tag) => {
+            if (AEM_TAG_PATTERN.test(value) === false) return false;
+            const [namespace, path] = tag.split(':');
+            if (!namespace || !path) return '';
+            return path ? `/content/cq:tags/${namespace}/${path}` : '';
+        })
+        .filter(Boolean);
+}
+
+/**
+ * Converts from property (path format) to attribute (tag format)
+ * @param {string} value Path format string (e.g., "/content/cq:tags/mas/product/photoshop")
+ * @returns {string} Tag format string (e.g., "mas:product/photoshop")
+ */
+export function toAttribute(value) {
+    if (!value || value.length === 0) return '';
+    return value
+        .map((path) => {
+            const match = path.match(/\/content\/cq:tags\/([^/]+)\/(.+)$/);
+            return match ? `${match[1]}:${match[2]}` : '';
+        })
+        .filter(Boolean)
+        .join(',');
+}
+
+class AemTagPickerField extends LitElement {
+    static properties = {
+        baseUrl: { type: String, attribute: 'base-url' },
+        label: { type: String },
+        bucket: { type: String },
+        open: { type: Boolean },
+        value: {
+            type: Array,
+            converter: {
+                fromAttribute,
+                toAttribute,
+            },
+            reflect: true,
+        },
+        namespace: { type: String },
+        top: { type: String },
+        multiple: { type: Boolean },
+        hierarchicalTags: { type: Object, state: true },
+        selected: { type: String },
+        ready: { type: Boolean, state: true },
+    };
+
+    static styles = css`
+        :host {
+            display: flex;
+            align-items: center;
+        }
+
+        sp-tags {
+            width: 100%;
+            position: relative;
+        }
+
+        sp-sidenav {
+            min-height: 300px;
+            max-height: 50vh;
+        }
+
+        sp-popover {
+            margin-top: var(--margin-picker-top, 0px);
+        }
+    `;
+
+    #aem;
+
+    constructor() {
+        super();
+        this.baseUrl = document.querySelector(
+            'meta[name="aem-base-url"]',
+        )?.content;
+        this.bucket = null;
+        this.top = null;
+        this.multiple = false;
+        this.hierarchicalTags = new Map();
+        this.value = [];
+        this.#aem = null;
+        this.ready = false;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.#aem = new AEM(this.bucket, this.baseUrl);
+        this.loadTags();
```

Comment:
await?

Reply:
connectedCallback is not async, cannot do await but, I don't need to await neither here.